compiler/syntax/src/res_core.ml

```diff
@@ -4080,12 +4080,19 @@ and parse_atomic_typ_expr ?current_type_name_path ?inline_types ~attrs p =
           ~constr_name:constr p
       in
       let number_of_inline_records_in_args =
-        args
-        |> List.filter (fun (c : Parsetree.core_type) ->
-               c.ptyp_attributes
-               |> List.exists (fun (({txt}, _) : Parsetree.attribute) ->
-                      txt = "res.inlineRecordReference"))
-        |> List.length
+        match inline_types with
+        | None -> 0
+        | Some inline_types ->
+          let inline_types = !inline_types in
+          args
+          |> List.filter (fun (c : Parsetree.core_type) ->
+                 match c.ptyp_desc with
+                 | Ptyp_constr ({txt = Lident typename}, _)
+                   when String.contains typename '.' ->
```

Comment:
This check that it contains `.` is unnecessary right? What counts is that it appears in `inline_types` which happen to be of that form right now.

Reply:
Done here: https://github.com/rescript-lang/rescript/pull/7241/commits/45286011bcb68266ce3f705b1fb02e053a3a3224